web/packages/teleport/src/components/ReAuthenticate/useReAuthenticate.ts

```diff
@@ -14,33 +14,44 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from 'react';
-
 import useAttempt from 'shared/hooks/useAttemptNext';
 
 import cfg from 'teleport/config';
 import auth from 'teleport/services/auth';
 
-export default function useReAuthenticate({
-  onAuthenticated,
-  onClose,
-  actionText = defaultActionText,
-}: Props) {
+import type { MfaAuthnResponse } from 'teleport/services/mfa';
+
+export default function useReAuthenticate(props: Props) {
+  const { onClose, actionText = defaultActionText } = props;
   const { attempt, setAttempt, handleError } = useAttempt('');
```

Comment:
How does `setAttempt` ever get set to `success` here?

Reply:
it doesn't or rather it's not necessary. i think this component was created in such that after a successful "submit" the control is passed back to the parent (with the `onAuthenticate` or `onMfaReponse` func). I thought it made sense b/c there shouldn't be anything else that needs to be done after user submits. Perhaps it would've made more sense to call `onClose` before `onAuthenticate/onMfaResponse`. So if we are closing the dialog, it's clearer that we don't need anything to re-render, which is the purpose of `setAttempt(success)`

Reply:
Maybe worth a concise comment explaining this because it's not the standard usage pattern.